### ISSUE TITLE
Sinatra 2.0.6 regression: OptionParser::InvalidOption:   invalid option: --pattern

### ISSUE THREAD

**Issue by rubys** [sinatra/sinatra#1559]
Error:

    OptionParser::InvalidOption:
      invalid option: --pattern
    # /var/lib/gems/2.5.0/gems/sinatra-2.0.6/lib/sinatra/main.rb:18:in `<module:Sinatra>'

Reproduction instructions:

$ git clone https://github.com/rubys/sinatra206.git
$ cd sinatra206
$ bundle install
$ bundle exec rake test

This worked with Sinatra 2.0.5.  See also https://travis-ci.org/apache/whimsy/jobs/574961311

**Comment by namusyaka** [comment]
Thanks for the report. We'll cut a new release v2.0.6.1 after fixing the issue.

**Comment by jgarber623** [comment]
I believe this same bug is causing issues with RSpec when invoked using `bundle exec rake`:

```sh
± bundle exec rake
/Users/jason/.rbenv/versions/2.6.3/bin/ruby -I/Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/rspec-core-3.8.2/lib:/Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/rspec-support-3.8.2/lib /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/rspec-core-3.8.2/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb

An error occurred while loading spec_helper.
Failure/Error: require File.expand_path('../config/environment', __dir__)

Bundler::GemRequireError:
  There was an error while trying to load the gem 'sinatra'.
  Gem Load Error is: invalid option: --pattern
  Backtrace for gem load error is:
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/sinatra-2.0.6/lib/sinatra/main.rb:18:in `<module:Sinatra>'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/sinatra-2.0.6/lib/sinatra/main.rb:1:in `<top (required)>'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/sinatra-2.0.6/lib/sinatra.rb:1:in `require'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/sinatra-2.0.6/lib/sinatra.rb:1:in `<top (required)>'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler/runtime.rb:81:in `require'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler/runtime.rb:81:in `block (2 levels) in require'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler/runtime.rb:76:in `each'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler/runtime.rb:76:in `block in require'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler/runtime.rb:65:in `each'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler/runtime.rb:65:in `require'
  /Users/jason/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/bundler-2.0.2/lib/bundler.rb:114:in `require'
# etc. etc. etc.
```

Strangely (to me, at least), `bundle exec rspec` works just fine which led me to the `--pattern` bit maybe being the issue.

`bundle exec rake` worked as expected before updating several of my applications to v2.0.6.

**Comment by namusyaka** [comment]
This is because the difference between the actual executable file and the actual caller haven't been handled properly.

**Comment by jgarber623** [comment]
Oh, shoot. Apologies for the noise. The application code linked in the initial bug report is also using RSpec.

**Comment by namusyaka** [comment]
@jgarber623 @rubys Could you try again with #1560?

**Comment by rubys** [comment]
My tests pass with #1560 applied.

**Comment by namusyaka** [comment]
I'll cut a patch release tonight (JST)
LMK if you still have issues.

**Comment by sue445** [comment]
@namusyaka thank you for bugfix. #1560 works on my app too!

**Comment by namusyaka** [comment]
Released v2.0.7: https://rubygems.org/gems/sinatra/versions/2.0.7
Thanks all for your feedback, it was very helpful to me.
